basic/dapps/universal-provider-solana/src/utils/helpers.ts

```diff
@@ -0,0 +1,175 @@
+import {
+  Connection,
+  PublicKey,
+  SystemProgram,
+  Transaction,
+  VersionedTransaction,
+  clusterApiUrl,
+} from "@solana/web3.js";
+import UniversalProvider from "@walletconnect/universal-provider/dist/types/UniversalProvider";
+import bs58 from "bs58";
+import nacl from "tweetnacl";
+
+export enum SolanaChains {
+  MainnetBeta = "4sGjMW1sUnHzSxGspuhpqLDx6wiyjNtZ",
+  Devnet = "8E9rvCKLFQia2Y35HXjjpWzj8weVo44K",
+}
+
+export function verifyTransactionSignature(
+  address: string,
+  signature: string,
+  tx: Transaction
+) {
+  return nacl.sign.detached.verify(
+    tx.serializeMessage(),
+    bs58.decode(signature),
+    bs58.decode(address)
+  );
+}
+
+export function verifyMessageSignature(
+  address: string,
+  signature: string,
+  message: string
+) {
+  return nacl.sign.detached.verify(
+    bs58.decode(message),
+    bs58.decode(signature),
+    bs58.decode(address)
+  );
+}
+
+const isVersionedTransaction = (
+  transaction: Transaction | VersionedTransaction
+): transaction is VersionedTransaction => "version" in transaction;
+
+
+export const getProviderUrl = (chainId: string) => {
```

Comment:
Did we confirm that our rpc supports solana?

Reply:
Yep, its [supported](https://github.com/WalletConnect/blockchain-api/blob/master/SUPPORTED_CHAINS.md#http-rpc) 🖖